assets/js/actions/respondentGroups.js

```diff
@@ -102,6 +102,10 @@ const handleRespondentGroupUpload = (dispatch, promise, groupId = null) => {
         } else {
           dispatch(receiveInvalids(value))
         }
+      },
+      () => {
+        // Reject with original error
+        Promise.reject(e)
```

Comment:
In my tests the JSON error didn't bubble up high enough, it logged on the browser console but no feedback to the user. The outer error did show the popup with not much information. It could have been replaced by `new Error("some message")`, still not very friendly.